search/v2/geocoder.cpp

```diff
@@ -1021,7 +1021,7 @@ void Geocoder::WithPostcodes(TFn && fn)
         break;
 
       TokenSlice slice(m_params, startToken, startToken + n);
-      if (LooksLikePostcode(slice))
+      if (LooksLikePostcode(QuerySliceOnTokens(slice), true /*handleAsPrefix*/))
```

Comment:
This is not true. isPrefix should be true if and only if startToken + n == m_numTokens. Please, add a test on this case.